lib/ansible/plugins/action/ironware.py

```diff
@@ -0,0 +1,79 @@
+# (c) 2016 Red Hat Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import sys
+import copy
+import json
+
+from ansible import constants as C
+from ansible.plugins.action.normal import ActionModule as _ActionModule
+from ansible.module_utils.network_common import load_provider
+from ansible.module_utils.ironware import ironware_provider_spec
+from ansible.module_utils.six import iteritems
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class ActionModule(_ActionModule):
+
+    def run(self, tmp=None, task_vars=None):
+
+        if self._play_context.connection != 'local':
+            return dict(
+                failed=True,
+                msg='invalid connection specified, expected connection=local, '
+                    'got %s' % self._play_context.connection
+            )
+
+        provider = load_provider(ironware_provider_spec, self._task.args)
+
+        pc = copy.deepcopy(self._play_context)
+        pc.connection = 'network_cli'
+        pc.network_os = 'ironware'
+        pc.remote_addr = provider['host'] or self._play_context.remote_addr
+        pc.port = int(provider['port'] or self._play_context.port or 22)
+        pc.remote_user = provider['username'] or self._play_context.connection_user
+        pc.password = provider['password'] or self._play_context.password
+        pc.private_key_file = provider['ssh_keyfile'] or self._play_context.private_key_file
+        pc.timeout = int(provider['timeout'] or C.PERSISTENT_COMMAND_TIMEOUT)
+        pc.become = provider['authorize'] or False
+        pc.become_pass = provider['auth_pass']
+
+        display.vvv('using connection plugin %s' % pc.connection, pc.remote_addr)
+        connection = self._shared_loader_obj.connection_loader.get('persistent', pc, sys.stdin)
+
+        socket_path = connection.run()
+
+        display.vvvv('socket_path: %s' % socket_path, pc.remote_addr)
+        if not socket_path:
+            return {'failed': True,
+                    'msg': 'unable to open shell. Please see: ' +
+                           'https://docs.ansible.com/ansible/network_debug_troubleshooting.html#unable-to-open-shell'}
+
+        task_vars['ansible_socket'] = socket_path
+
```

Comment:
In other modules that support `authorize` we have some code here to check we are in the right context (privileged shell vs non-privileged).

If you were to test the code with a task that requires privileged then one that requires non-privileged then I believe the current code would fail.

`ios` supports `authorize` as well: 
https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/ios.py#L72-L86

Reply:
@gundalow I'm not sure this code will detect priv vs non-priv connection - it looks like it detects whether we have somehow landed in config mode vs priv/enable mode? Is this somehow possible with the persistent connection framework if we connect to a socket previously used by a config job that died part way through? If so, agree this will be required.

Does the code here actually catch the case when the user is expecting to be in enable mode but isn't? https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/terminal/ios.py#L66

Reply:
Also, can you explain what these lines achieve? 
https://github.com/ansible/ansible/blob/88da95bb77cf20f56035e7ba5d0230c0a2785f2b/lib/ansible/plugins/action/ios.py#L81-L83